File: product-backend/src/main/java/org/kainos/ea/service/CapabilityValidator.java
package org.kainos.ea.service;

import org.kainos.ea.model.CapabilityRequest;

public class CapabilityValidator {
    public String isCapabilityValid(CapabilityRequest capability) {

        if (capability.getCapabilityName().length() > 50) {
            return "Capability name longer than 50 chars";
        }
        if (capability.getLeadName().length() > 50) {
            return "Capability lead name longer than 50 chars";
        }
        if (capability.getCapabilityLeadPicture().length() > Integer.MAX_VALUE - 1) {
            return "Lead picture too large";
        }
        if (capability.getMessage().length() > 3000) {
            return "Message too long";
        }
        return null;
    }
}

File: product-web-app/service/capabilityValidator.ts
import Capability from '../model/capability.js';

export default class CapabilityValidator {
  validateCapability(capability: Capability) {
    if (capability.capabilityName.length > 50) {
      return 'Capability name greater than 50 characters';
    }

    if (capability.leadName.length > 50) {
      return 'Lead name greater than 50 characters';
    }

    if (capability.capabilityLeadPicture.length > 2147483647 - 1) {
      return 'Picture size too large';
    }
    if (capability.message.length > 3000) {
      return 'Message too long';
    }

    return null;
  }
}
